components/collator/src/provider.rs

```diff
@@ -539,6 +559,15 @@ pub struct CollationSpecialPrimaries<'data> {
     /// variants in `MaxVariable`, currently 4.
     #[cfg_attr(feature = "serde", serde(borrow))]
     pub last_primaries: ZeroVec<'data, u16>,
+    /// 256 bits to classify every possible byte
+    /// into compressible or non-compressible.
+    /// The 256 bits are distributed across 32
+    /// `u8`s, since we don't have a native `u256`
+    /// and have to do manual shifts and masks,
+    /// so we might as well do them with the narrow
+    /// type that is its own `ULE`.
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub compressible_bytes: ZeroVec<'data, u8>,
```

Comment:
Suggestion (optional): Since this struct now contains 2 fixed-length ZeroVecs, it would be slightly more efficient to represent them as a single ZeroVec, or possibly a MultiFieldsULE.

Reply:
The other `ZeroVec` isn't logically fixed-length for all time, but it has 4 items for now and has had 4 items for a long time.

Also, the other one has `u16`s, and I felt uneasy about introducing bugs to the old thing if I tweak it in a hurry.

Reply:
> which isn't much more efficient than just using a `[u8; 32]` here.

Are we allowed to use `[u8; 32]` directly in a data struct?

Reply:
Yes, you can use any stack type in a data struct, it's only heap types like strings and vectors where you need ZeroVec.

The stack types mostly need to implement Bake. Sometimes there's some Yoke stuff.

Reply:
Observation: if you can get the compressible bytes to fit in `[u8; 31]` instead of `[u8; 32]`, then it can fit into the padding of the `CollationSpecialPrimaries` and the stack size is equal to what it would be if you added another ZeroVec.

Reply:
Unfortunately, I don't know enough of what guarantees FractionalUCA/genuca make to know whether `[u8; 31]` would work.

Reply:
In the borrowed version it's behind a reference. In the owned version, the `DataPayload` contains the data struct (plus an RC to the buffer).

Reply:
> Unfortunately, I don't know enough of what guarantees FractionalUCA/genuca make to know whether `[u8; 31]` would work.

Currently, the last item is non-zero and first item is zero.